src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useState } from 'react'
+import { useVideosOffsetLimitPagination } from '@/api/hooks'
+import { useDrafts } from '@/hooks'
+import { VideoPreviewPublisher } from '@/components'
+import { Grid, Pagination, Tabs } from '@/shared/components'
+
+import {
+  PaginationContainer,
+  StyledDismissibleMessage,
+  StyledText,
+  TabsContainer,
+  ViewContainer,
+} from './MyVideos.styles'
+import { AddVideo, AddVideoView } from './AddVideoView'
+
+// const testChannelId = 'a49fc01c-d369-44d2-b272-bcf0b0d26a5e' // mocking test channel id
+const testChannelId = '100' // staging test channel id
+const TABS = ['All Videos', 'Published', 'Drafts', 'Unlisted'] as const
+const INITIAL_VIDEOS_PER_ROW = 4
+// not yet doable
+// TODO: on edit video callbacks
+// TODO: on delete video callbacks
+// TODO: dynamic channels (not hardcoded)
+export const MyVideosView = () => {
+  const [videosPerRow, setVideosPerRow] = useState(INITIAL_VIDEOS_PER_ROW)
+  const [hasAnyVideos, sethasAnyVideos] = useState<boolean>()
+  // Drafts calls can run into race conditions
+  const { drafts, removeDraft, removeAllDrafts, addDraft } = useDrafts('video', testChannelId)
+  const [currentTab, setCurrentTab] = useState(0)
+  const { currentPage, setCurrentPage } = usePagination(currentTab)
+  const videosPerPage = 4 * videosPerRow
+  const currentTabName = TABS[currentTab]
+  const isPublic_eq = getPublicness(currentTabName)
+  const { loading, videos, error, totalCount, fetchMore } = useVideosOffsetLimitPagination(
+    {
+      limit: videosPerPage,
+      offset: videosPerPage * (currentPage - 1),
+      where: {
+        channelId_eq: testChannelId,
+        isPublic_eq,
+      },
+    },
+    {
+      // notifyOnNetworkStatusChange: true,
+      fetchPolicy: 'cache-first',
+    }
+  )
+
+  useEffect(() => {
+    if (typeof totalCount === 'number' && totalCount > 0) {
+      sethasAnyVideos(true)
+    } else if (typeof totalCount === 'number' && hasAnyVideos === undefined) {
+      sethasAnyVideos(false)
+    }
+  }, [hasAnyVideos, totalCount])
+
+  // hook to tests draft should be deleted before final
+  useEffect(() => {
+    const createDrafts = async () => {
+      await removeAllDrafts()
+      for (let i = 0; i < 32; i++) {
+        await addDraft({
+          channelId: testChannelId,
+          title: i.toString(),
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
```

Comment:
When would that second condition kick in?

Reply:
When calling `fetchMore()` on the pagination it sets videos to `[]` while it gets the new data